third_party/amd/lib/TritonAMDGPUTransforms/Utility.cpp

```diff
@@ -0,0 +1,96 @@
+#include "Utility.h"
+
+#include "mlir/Dialect/SCF/IR/SCF.h"
+
+namespace findMin {
+int findMinInBlock(Block &block,
+                   const std::function<int(Operation *)> &countFunc);
+
+// Returns the minimum found when accumulating countFunc(op) for all paths
+// between op1 and op2
+int findMinBetweenOps(Operation *op1, Operation *op2,
```

Comment:
Nit: `beginOp` and `endOp` to be more clear. Generally it's better to have self-documenting parameter names.